File: app/src/main/java/com/iug/coursehub/feature/dashboard/DashboardNavHostViewModel.java
package com.iug.coursehub.feature.dashboard;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class DashboardNavHostViewModel extends ViewModel {

    private final MutableLiveData<UiAction> _uiAction = new MutableLiveData<>();
    private final LiveData<UiAction> uiAction = _uiAction;

    public LiveData<UiAction> getUiAction() {
        return uiAction;
    }

    public void navigateToAuthScreen() {
        _uiAction.postValue(UiAction.NAVIGATE_TO_AUTH_NAV_HOST_SCREEN);
    }

    public enum UiAction {
        NAVIGATE_TO_AUTH_NAV_HOST_SCREEN
    }
}

File: app/src/main/java/com/iug/coursehub/feature/dashboard/DashboardNavHostViewModel.kt
package com.iug.coursehub.feature.dashboard

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class DashboardNavHostViewModel : ViewModel() {

    private val _uiAction = MutableLiveData<UiAction>()
    val uiAction: LiveData<UiAction> = _uiAction

    fun navigateToAuthScreen() {
        _uiAction.postValue(UiAction.NavigateToAuthNavHostScreen)
    }

    sealed interface UiAction {
        data object NavigateToAuthNavHostScreen : UiAction
    }
}